### PR TITLE
bootstrap command

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. add `rocket bootstrap` command which scaffolds files for a basic rocket site
2. add `codeTabs` preset to docs site
3. improve Getting Started and Presets Guides
4. added list of community presets and link to edit them
5. adds type inference to `addPlugin` => https://tsplay.dev/wXknVW this might not work with js sources though.
6. adds indexes to docs sections
7. improves presets docs
8. hides search bar for noscript users
9. fixes noscript bug for our site
10. adds markdown odd table row colour
11. adds styling docs for search
11. added noscript styles for `<inline-notification>`

### CODE REVIEW

**Comment by bennypowers** [comment]
One of the best features in this PR is type inference in `addPlugin`. If the plugin function is typed, so is the `options` property. in this case, the options object is an object with values that take a unified transformer's params.
Screen Shot 2021-07-09 at 1 19 47

File: docs/docs/presets/joining-blocks.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+# Presets >> Joining Blocks || 10
 
 The template system allows for a very granular control of how individual parts will be merged, overwritten or reorderd.
 
 As a preset you may want to add this to your layout.
 
 {% raw %}
 
-```
+```jinja2
```
Comment (daKmoR) [inline docs/docs/presets/joining-blocks.md:9]: huh - what is that? I'm curious 🤗

**Comment by daKmoR** [comment]
wwwoooooowwww that PR is amazing 🤩

sooo much goodness in here 💪

thank you 🙇‍♂️

File: docs/docs/presets/joining-blocks.md (hunk 1 above)
Comment (bennypowers) [reply docs/docs/presets/joining-blocks.md:9]: refractor (the syntax highlighting lib) doesn't have njk, but as nunjucks is a fork of jinja, jinja2 was the closest available alternative.

**Pull request merged**